src/jattach/jattach_openj9.c

```diff
@@ -137,7 +139,13 @@ static int read_response(int fd, const char* cmd, int print_output) {
         }
 
         if (off >= size) {
-            buf = realloc(buf, size *= 2);
+            char* temp_buf = realloc(buf, size *= 2);
+            if (temp_buf == NULL) {
+                free(buf);
```

Comment:
I think `realloc` is responsible for freeing the old buffer if needed, am I missing something?